Feature/relnotes (#215)
Avaiga/taipy-doc: docs/relnotes.md

This is the list of changes to Taipy releases as they were published.

> ## Community edition: 2.0 (In progress)

Reviewer: Why the (in progress) segment?

Author: Cause it s not delivered yet. But I can remove it.

Published on TO_BE_DEFINED.

[`taipy` 2.0](https://pypi.org/project/taipy/2.0.0/) contains the latest

> [`taipy-config` 1.0](https://pypi.org/project/taipy-config/1.0.0/),

Reviewer: I thought we said... 2.0?

Author: yep

[`taipy-gui` 2.0](https://pypi.org/project/taipy-gui/2.0.0/),
[`taipy-core` 2.0](https://pypi.org/project/taipy-core/2.0.0/) and
[`taipy-rest` 2.0](https://pypi.org/project/taipy-rest/2.0.0/) packages.

#### Features

**`taipy-core`** 2.0.0

   - TODO SQLDataNodes
   - TODO SQLTableDatanode
   - TODO JSONDataNodes
   - In standalone mode, a job whose status is `SUBMITTED`, `PENDING`, or `BLOCKED`, can be canceled. When
     canceling the job, its subsequent jobs will be abandoned and their statuses will be set to `ABANDONED`. When
     the cancel method is called on a job with its status being either `RUNNING`, `COMPLETED`, or `SKIPPED`, its
     subsequent jobs will be abandoned while its status remains unchanged. A job whose status is `FAILED`,

>      `CANCELED`, or `ABANDONED`, cannot be canceled.

Reviewer: I would have removed the comma before 'cannot'.
And rewrite 'cannot' to 'can not'.

Author: Good question about cannot.
I just found [this](https://www.grammarly.com/blog/cannot-or-can-not/#:~:text=Here%20is%20a%20quick%20summary,.%20.%20.%20but%20also.%E2%80%9D) on grammarly.
I would keep the cannot.

Reviewer: See https://www.merriam-webster.com/words-at-play/cannot-vs-can-not-is-there-a-difference#:~:text=Both%20cannot%20and%20can%20not,general%2C%20it%20is%20somewhat%20informal.

"Cannot has been in use since the 15th century"... 
Cannot is the way to go. Please forget my comment.

   - TODO Core as a service

#### Improvements

> **`taipy-core`** 2.0.0

Reviewer: Ok so you did not like my carriage return... okok

Author: For me, the title is the name of the versioned package. The "core" and "2.0" terms come together. I don't see the purpose of separating them with a carriage return. But well, that is minor. If you are convinced we need it, I will revert without any ressentiment.
:)

   - The data node of a scenario or a pipeline can now be accessed directly at the scenario or pipeline levels.
   - When submitting a scenario, a pipeline or a task, a list of jobs or a Job will be returned.
   - When submitting a scenario, pipeline, or task in standalone mode, the user can use the parameters _wait_ and
     _timeout_ to wait until the submitted jobs are finished or up to _timeout_ seconds.
   - When a running job is failed, its subsequent jobs will be abandoned.
   - A primary scenario can be deleted along with its cycle if it is the only scenario in the cycle.
   - TODO Exception messages

**`taipy-config`** 1.0.0

   - TODO New package

#### Significant bug fixes

#### Deprecations

   - TODO Config field nb_of_workers into max_nb_of_workers

> ## Enterprise edition: 2.0 (In progress)

Reviewer: "In progress"?

Author: removed

This is the list of changes to Taipy releases as they were published.

## Community edition: 2.0

Published on TO_BE_DEFINED.

[`taipy` 2.0](https://pypi.org/project/taipy/2.0.0/) contains the latest
[`taipy-config` 2.0](https://pypi.org/project/taipy-config/2.0.0/),
[`taipy-gui` 2.0](https://pypi.org/project/taipy-gui/2.0.0/),
[`taipy-core` 2.0](https://pypi.org/project/taipy-core/2.0.0/) and
[`taipy-rest` 2.0](https://pypi.org/project/taipy-rest/2.0.0/) packages.

#### Features

**`taipy-core`** 2.0.0

>    - New data node: SQLTableDataNode. It represents a table in a SQL database.

Reviewer: "New data node: SQLTableDataNode. It represents a table in a SQL database."
->
"New data node class: `SQLTableDataNode`. It represents a table in an SQL database.`

Same remark for JSON (not the 'an' thing of course)

   - New data node: JSONDataNode. It represents the data from a JSON file.
   - SQLDataNode behavior is changed due to the release of SQLTableDataNode. Now it represents the data
     using custom read and write queries.
   - In standalone mode, a job whose status is `SUBMITTED`, `PENDING`, or `BLOCKED`, can be canceled. When
     canceling the job, its subsequent jobs will be abandoned and their statuses will be set to `ABANDONED`. When
     the cancel method is called on a job with its status being either `RUNNING`, `COMPLETED`, or `SKIPPED`, its
     subsequent jobs will be abandoned while its status remains unchanged. A job whose status is `FAILED`,
     `CANCELED`, or `ABANDONED` cannot be canceled.

>    - Taipy Core can now be run as a service by using `Core().run()` or `tp.run(Core())`. By running Core as a service, Taipy will initialize the scheduler and the job dispatcher based on the config provided. Taipy Core can also be run along with Taipy Gui or Taipy Rest.

Reviewer: 'Taipy will initialize' -> 'Taipy initializes'
'config provided;  -> 'provided configuration'
'Taipy Core can also be run along with Taipy Gui or Taipy Rest' ->
  'The Taipy Core service can be run along with Taipy Gui or Taipy Rest services' ->

#### Improvements

**`taipy-core`** 2.0.0

   - The data node of a scenario or a pipeline can now be accessed directly at the scenario or pipeline levels.

>    - When submitting a scenario, a pipeline or a task, a list of jobs or a Job will be returned.

Reviewer: Job -> `Job`

   - When submitting a scenario, pipeline, or task in standalone mode, the user can use the parameters _wait_ and
     _timeout_ to wait until the submitted jobs are finished or up to _timeout_ seconds.

>    - When in standalone mode, the job dispatcher will be running in a sub-thread that will periodically check for new jobs submitted by Taipy to execute.

Reviewer: 'the job dispatcher will be running in a sub-thread that will periodically check'
  -> 
  'the job dispatcher runs in a sub-thread that periodically checks'

   - When a running job is failed, its subsequent jobs will be abandoned.
   - A primary scenario can be deleted along with its cycle if it is the only scenario in the cycle.

>    - The messages of the various Exceptions that can be raised have been improved to help the users debug their

Reviewer: 'Exceptions' -> 'exceptions'

     applications.

**`taipy-config`** 1.0.0

   - TODO New package

#### Significant bug fixes

#### Deprecations

>    - The field nb_of_workers within the Config has been deprecated in favor of max_nb_of_workers.

Reviewer: 'nb_of_workers' -> '*nb_of_workers*'
(same fox max*)

## Enterprise edition: 2.0

Published on TO_BE_DEFINED.

#### Features

>    - New repository: SQLLite can now be used as a replacement for filesystem as the taipy repository.
>    - New repository: MongoDB can now be used as a replacement for filesystem as the taipy repository. 

Author: ```suggestion
   - SQLLite or MongoDB databases can now be used as alternatives to the filesystem to store Taipy entities.
```

This is the list of changes to Taipy releases as they were published.

## Community edition: 2.0

Published on TO_BE_DEFINED.

[`taipy` 2.0](https://pypi.org/project/taipy/2.0.0/) contains the latest
[`taipy-config` 2.0](https://pypi.org/project/taipy-config/2.0.0/),
[`taipy-gui` 2.0](https://pypi.org/project/taipy-gui/2.0.0/),
[`taipy-core` 2.0](https://pypi.org/project/taipy-core/2.0.0/) and
[`taipy-rest` 2.0](https://pypi.org/project/taipy-rest/2.0.0/) packages.

#### Features

**`taipy-core`** 2.0.0

>    - New data node named SQLTableDataNode. It represents a table in a SQL database.

Author: ```suggestion
   - New data node named `SQLTableDataNode`. It represents a table in an SQL database.
```

>    - New data node named JSONDataNode. It represents the data from a JSON file.

Author: ```suggestion
   - New data node named `JSONDataNode`. It represents the data from a JSON file.
```

   - SQLDataNode behavior is changed due to the release of SQLTableDataNode. Now it represents the data
     using custom read and write queries.
   - In standalone mode, a job whose status is `SUBMITTED`, `PENDING`, or `BLOCKED` can be canceled. When canceling
     the job, its subsequent jobs will be abandoned, and their statuses will be set to `ABANDONED`. When the cancel
     method is called on a job whose status is either `RUNNING`, `COMPLETED`, or `SKIPPED`, its subsequent jobs will
     be abandoned while its status remains unchanged. A job whose status is `FAILED`, `CANCELED` or `ABANDONED`
     cannot be canceled.
   - Taipy Core can now be run as a service by using `Core().run()` or `tp.run(Core())`. By running Core
     as a service, Taipy will initialize the scheduler and the job dispatcher based on the config provided. Taipy

>      Core can also be run along with Taipy Gui or Taipy Rest.

Author: ```suggestion
   - Taipy Core can now be run as a service by using `Core().run()` or `tp.run(Core())`. By running Core
     as a service, Taipy initializes the scheduler and the job dispatcher based on the provided configuration. The Taipy Core service can be run along with Taipy Gui or Taipy Rest services.
```

#### Improvements

**`taipy-core`** 2.0.0

   - The data node of a scenario or a pipeline can now be accessed directly at the scenario or pipeline
     levels.
   - When submitting a scenario, a pipeline, or a task, the job(s) created will be returned.
   - When submitting a scenario, pipeline, or task in standalone mode, the user can use the parameters
     _wait_ and _timeout_ to wait until the submitted jobs are finished or up to _timeout_ seconds.
   - When in standalone mode, the job dispatcher will be running in a sub-thread that will periodically

>      check for new jobs submitted by Taipy to execute.

Author: ```suggestion
   - When in standalone mode, the job dispatcher runs in a sub-thread that periodically checks for new jobs submitted by Taipy to execute.
```

   - When a running job fails, its subsequent jobs will be automatically abandoned.
   - A primary scenario can be deleted along with its cycle if it is the only scenario in the cycle.

>    - The messages of the various Exceptions that can be raised have been improved to help the users

Author: ```suggestion
   - The messages of the various exceptions that can be raised have been improved to help the users
```

     debug their applications.

**`taipy-config`** 2.0.0

   - New taipy-config package made to be used by any other Taipy package for configuration and logging.

#### Significant bug fixes

#### Deprecations

**`taipy-core`** 2.0.0

>    - The field nb_of_workers within the Config has been deprecated in favor of max_nb_of_workers.

Author: ```suggestion
   - The field `*nb_of_workers*` within the Config has been deprecated in favor of `*max_nb_of_workers*`.
```

#### Features

>    - New repository: SQLLite can now be used as a replacement for filesystem as the taipy repository.

Reviewer 2: @jrobinAV @FabienLelaquais Any feedback on the text is apreciated. Cheers.

Author: I would group both entries, and I would not use the term repository which can be misinterpreted.

Reviewer: And I would capitalize 'Taipy'
